### PR TITLE
Share method handle related intrinsics

### PR DESCRIPTION
**Pull request by vitek-karas**
Moving some simple method handle related intrinsics and unifying the values related to it.

### CODE REVIEW

File: test/Mono.Linker.Tests.Cases/DataFlow/MakeGenericDataFlow.cs [hunk 1]
```diff
@@ -684,6 +686,12 @@ static void TestWithUnmanagedConstraint ()
 			{
 				var t = new T ();
 			}
+
+			static void TestGetMethodFromHandle (Type unknownType = null)
+			{
+				MethodInfo m = (MethodInfo) MethodInfo.GetMethodFromHandle (typeof (MakeGenericMethod).GetMethod (nameof (GenericWithNoRequirements)).MethodHandle);
+				m.MakeGenericMethod (unknownType);
```
Comment (sbomer) [inline test/Mono.Linker.Tests.Cases/DataFlow/MakeGenericDataFlow.cs:693]: Just making sure I understand how the test works - is the idea that if we didn't support GetMethodFromHandle, the call to `MakeGenericMethod` would warn about an unknown method?

Might be worth adding a test that produces some warnings (where the generic parameters have requirements).

File: test/ILLink.RoslynAnalyzer.Tests/generated/ILLink.RoslynAnalyzer.Tests.Generator/ILLink.RoslynAnalyzer.Tests.TestCaseGenerator/UnreachableBlockTests.g.cs [hunk 2]
```diff
@@ -37,6 +37,12 @@ public Task InstanceMethodSubstitutions ()
 			return RunTest (allowMissingWarnings: true);
 		}
 
+		[Fact]
+		public Task MethodArgumentPropagation ()
```
Comment (sbomer) [inline test/ILLink.RoslynAnalyzer.Tests/generated/ILLink.RoslynAnalyzer.Tests.Generator/ILLink.RoslynAnalyzer.Tests.TestCaseGenerator/UnreachableBlockTests.g.cs:41]: I think @tlakollo already added these

**Review by sbomer — APPROVED** [review]
LGTM, thank you!

File: test/Mono.Linker.Tests.Cases/DataFlow/MakeGenericDataFlow.cs (hunk 1 above)
Comment (vitek-karas) [reply test/Mono.Linker.Tests.Cases/DataFlow/MakeGenericDataFlow.cs:693]: Yes - if the method is not know MakeGenericMethod should always warn... so the fact that it doesn't warn means we were able to correctly flow the MethodInfo.
I'll add a negative test as well.

**Pull request merged**